shung93/portfolio
# Tidy project structure and file names

## Background
This PR is to introduce React naming and file extension conventions: 
- use the `.jsx` file extension for files that contain JSX
- use pascal case for component filenames
- prepend `use` to the hooks' names.

## What's changed
- updated all components to use pascal case for the filename and use the `.jsx` extension
- update the `isVisible` hook name and filename to `useVisibility`
- renamed export `IndexPage` component in `pages/contact` to `ContactPage` to match the actual page
- moved `ScrollProgress` and `ToggleContent` from the hooks sub-directory into the parent `components` directory since they are being used as components rather than hooks
- updated imports to reflect name changes
- other miscellaneous formatting updates via Prettier

## How to test
This PR doesn't have any UI changes and everything should be working as before

1. Pull a copy of the PR and run locally using `npm run develop`
2. Use the site and make sure everything is working as expected
